lib/flagsmith/engine/features/models.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+require_relative '../utils/hash_func'
+
+module Flagsmith
+  module Engine
+    # FeatureModel
+    class Feature
+      attr_reader :id, :name, :type
+
+      def initialize(id:, name:, type:)
+        @id = id
+        @name = name
+        @type = type
+      end
+
+      def ==(other)
+        return false if other.nil?
+
+        id == other.id
+      end
+
+      class << self
+        def build(json)
+          new(**json.slice(:id, :name, :type))
+        end
+      end
+    end
+
+    module Features
+      # MultivariateFeatureOptionModel
+      class MultivariateOption
+        attr_reader :value, :id
+
+        def initialize(value:, id: nil)
+          @value = value
+          @id = id
+        end
+
+        class << self
+          def build(json)
+            new(**json.slice(:id, :value))
+          end
+        end
+      end
+
+      # MultivariateFeatureStateValueModel
+      class MultivariateStateValue
+        attr_reader :id, :multivariate_feature_option, :percentage_allocation, :mv_fs_value_uuid
+
+        def initialize(id:, multivariate_feature_option:, percentage_allocation:, mv_fs_value_uuid: SecureRandom.uuid)
+          @id = id
+          @percentage_allocation = percentage_allocation
+          @multivariate_feature_option = multivariate_feature_option
+          @mv_fs_value_uuid = mv_fs_value_uuid
+        end
+
+        def <=>(other)
+          return false if other.nil?
+
+          if !id.nil? && !other.id.nil?
+            id - other.id
+          else
+            mv_fs_value_uuid <=> other.mv_fs_value_uuid
+          end
+        end
+
+        class << self
+          def build(json)
+            new(
+              **json.slice(:id, :percentage_allocation, :mv_fs_value_uuid)
+                    .merge(multivariate_feature_option: MultivariateOption.build(json[:multivariate_feature_option]))
+            )
+          end
+        end
+      end
+
+      # FeatureSegment
+      class Segment
```

Comment:
Maybe it's sufficient to use modules to namespace classes but it feels _slightly_ odd to me to have a Segment class here and in the segments module? Is this normal practice in Ruby?